### PR TITLE
Build MoM to Soak without release to public universe

### PR DESCRIPTION
**Pull request by kensipe**
Summary:
Build MoM to Soak without release to public universe

JIRA issues:  MARATHON-8363

### CODE REVIEW

File: ci/dcos.sc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * Creates DCOS package for Marathon
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param packagingTool the registry build tool
+ * @param version used to identify the correct build file
+ */
+
+def buildDCOS(packagingTool: Path, version: SemVer) {
+  // ./dcos-registry-darwin registry migrate --package-directory=repo --output-directory=target
+  Try(%(packagingTool, "registry", "migrate", s"--package-directory=$UNIVERSE_DIR", s"--output-directory=$PACKAGE_DIR"))
```
Comment (jeschkies) [inline ci/dcos.sc]: What is the reason for wrapping it in a `Try`?

File: dcos/resource.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (jeschkies) [inline dcos/resource.json]: So this is the new registry package builder config? Is this really a template?

File: ci/dcos.sc [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  buildDCOS(packagingTool, version)
+}
+
+/**
+ * Creates catalog package for Marathon with version info
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param version updated in the DCOS package files
+ */
+def createCatalogPackage(version: SemVer) {
```
Comment (jeschkies) [inline ci/dcos.sc]: What do you think about returning the path to the package instead of using a global constant?

File: ci/dcos.sc [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+ * @param packagingTool the registry build tool
+ * @param version used to identify the correct build file
+ */
+
+def buildDCOS(packagingTool: Path, version: SemVer) {
+  // ./dcos-registry-darwin registry migrate --package-directory=repo --output-directory=target
+  Try(%(packagingTool, "registry", "migrate", s"--package-directory=$UNIVERSE_DIR", s"--output-directory=$PACKAGE_DIR"))
+  val registryJSON = PACKAGE_DIR / s"marathon-${version.toReleaseString}.json"
+  // ./dcos-registry-darwin registry build --build-definition-file=target/marathon-1.7.50.json --output-directory=target --use-local
+  %(packagingTool, "registry", "build", s"--build-definition-file=$registryJSON", s"--output-directory=$PACKAGE_DIR", "--use-local")
```
Comment (jeschkies) [inline ci/dcos.sc]: How is `target/marathon-1.7.50.json` created? It says build definition. So is it something we provide or is it generated?

File: ci/dcos.sc [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+import $file.utils
+import utils.SemVer
+
+// output directory for all packages going to S3
+val PACKAGE_DIR: Path = pwd / 'target / 'universal
+// dcos catalog template directory
+val TEMPLATE_DIR: Path = pwd / 'dcos
+// versioned dcos catalog directory
+val UNIVERSE_DIR: Path = pwd / 'target / 'dcos
```
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: this is to be run from marathon root... the pwd will be marathon resulting in `./target/dcos`.  nothing is written under `/ci`

File: ci/dcos.sc [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * Creates catalog package for Marathon with version info
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param version updated in the DCOS package files
+ */
+def createCatalogPackage(version: SemVer) {
+  rm! UNIVERSE_DIR
+  mkdir! UNIVERSE_DIR
```
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: I was just keeping significant variables used for build structure in the same place.   I will change this based on feedback.

File: ci/dcos.sc [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+  // list all files in temp
+  TEMPLATE_DIR.toIO.listFiles.foreach(file => {
+    val catalogFile = UNIVERSE_DIR / file.getName
+
+    Source.fromFile(file).getLines.map {
+      x => if(x.contains("__FILL_IN_VERSION_HERE__")) x.replace("__FILL_IN_VERSION_HERE__", version.toReleaseString) else x }
+      .foreach(x => write.append(catalogFile, x + System.lineSeparator()))
+    }
```
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: debated that.. but this was easy and easy to read... it uses the same variable `__FILL_IN_VERSION_HERE__` in text that the mom-ee scripts use.

File: dcos/resource.json (hunk 2 above)
Comment (kensipe) [reply dcos/resource.json]: we need all the dcos catalog files.

File: ci/dcos.sc (hunk 3 above)
Comment (kensipe) [reply ci/dcos.sc]: sure

File: dcos/resource.json (hunk 2 above)
Comment (jeschkies) [reply dcos/resource.json]: Ok.

File: ci/dcos.sc (hunk 7 above)
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/dcos.sc]: Ah. Got it. Thanks for clarifying.

File: ci/dcos.sc (hunk 6 above)
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/dcos.sc]: I always try to pass everything necessary and use a global at the very beginning. We definitely should switch to Gradle 😉 which would take care of that.

File: ci/dcos.sc (hunk 5 above)
[the remark this answers is not in the document]
Comment (timcharper) [reply ci/dcos.sc]: I wonder if we should have a require statement

File: ci/dcos.sc [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+// output directory for all packages going to S3
+val PACKAGE_DIR: Path = pwd / 'target / 'universal
+// dcos catalog template directory
+val TEMPLATE_DIR: Path = pwd / 'dcos
+// versioned dcos catalog directory
+val UNIVERSE_DIR: Path = pwd / 'target / 'dcos
+
+@main
+def localBuild(version: String) {
```
[the remark this answers is not in the document]
Comment (timcharper) [reply ci/dcos.sc]: Let's use SemVer. We have a scopt parser for it.

[the remark this answers is not in the document]
Comment (timcharper) [reply ci/dcos.sc]: currently in `ci/pipeline`

```
implicit val SemVerRead: scopt.Read[SemVer] =
  scopt.Read.reads(SemVer(_))
```

Let's move to `utils` so it can be shared?

File: ci/dcos.sc [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+/**
+ * Main function for building DCOS registry file that works
+ * with DCOS 1.12+.
+ *
+ * @param version updated in the DCOS package files
+ */
+
+def buildRegistry(version: SemVer) {
```
Comment (timcharper) [inline ci/dcos.sc]: procedural syntax deprecated: https://contributors.scala-lang.org/t/proposal-to-remove-procedure-syntax/2143

Please use `: Unit  = {`

File: ci/dcos.sc (hunk 8 above)
Comment (timcharper) [inline ci/dcos.sc]: Procedure syntax

File: ci/dcos.sc (hunk 3 above)
Comment (timcharper) [reply ci/dcos.sc]: Procedure syntax is deprecated

File: ci/dcos.sc (hunk 9 above)
Comment (kensipe) [reply ci/dcos.sc]: roger that.. thx!

File: ci/dcos.sc (hunk 1 above)
Comment (timcharper) [reply ci/dcos.sc]: It seems like we should log, at least, if this fails; I'm not sure why we'd be okay with it failing to begin with.

Comment (kensipe) [reply ci/dcos.sc]: I'm removing it... but it would be nice to be consistent: https://github.com/mesosphere/marathon/blob/master/ci/pipeline#L94-L96

File: ci/dcos.sc (hunk 8 above)
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: perfect.. thanks @timcharper

File: ci/dcos.sc (hunk 5 above)
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: this is a script and not Scala code... but I have removed all global variable references

File: ci/dcos.sc (hunk 8 above)
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: This is completely removed

File: ci/dcos.sc [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  )
+}
+
+/**
+ * Pulls the registry build tool for the specific OS.
+ * Uses the marathon version which contains the "--use-local"
+ *
+ */
+
+def getPackagingTool(): Path = {
```
[the remark this answers is not in the document]
Comment (kensipe) [reply ci/dcos.sc]: I don't see a reason to move it there

**Comment by kensipe** [comment]
rebased to master with gitignore conflict resolution

**Comment by kensipe** [comment]
rebased to changes on master

File: ci/dcos.sc [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+  )
+  universePath
+}
+
+/**
+ * Pulls the registry build tool for the specific OS.
+ * Uses the marathon version which contains the "--use-local"
+ *
+ */
+def getPackagingTool(): Path = {
```
Comment (jeschkies) [inline ci/dcos.sc]: I really would make this part of the `./ci/provision` as that file includes everything the a host requires to run a build. The same is true for `./ci/si_provision`. Leaving it here and calling it in `buildRegistry` makes it unsafe to use.

File: ci/dcos.sc [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+/**
+ * Creates DCOS package for Marathon
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param packagingTool the registry build tool
+ * @param version used to identify the correct build file
+ */
+def buildDCOS(
```
Comment (jeschkies) [inline ci/dcos.sc]: Technically it's not building `DC/OS` 😉.

File: ci/dcos.sc [hunk 13]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env amm
+
+import ammonite.ops._
+import ammonite.ops.ImplicitWd._
+
+import scala.io.Source
+
+import $file.utils
+import utils.SemVer
+import utils.SemVerRead
```
Comment (jeschkies) [inline ci/dcos.sc]: Why not `import $file.utils.{SemVer, SemVerRead}`?

File: ci/pipeline [hunk 14]
```diff
@@ -231,6 +234,8 @@ def buildMaster(): Unit = {
   val buildNumber = sys.env.get("BUILD_NUMBER").getOrElse("0")
   val version = build(buildName = s"master-$buildNumber")
   buildDockerAndLinuxPackages()
+  testDockerAndLinuxPackages()
```
Comment (jeschkies) [inline ci/pipeline]: This seems to be an unrelated change. It will be introduced with #6550.

Comment (kensipe) [reply ci/pipeline]: I think this came with the rebase... I remove

File: ci/dcos.sc (hunk 12 above)
Comment (kensipe) [reply ci/dcos.sc]: definitely

File: ci/dcos.sc (hunk 11 above)
Comment (kensipe) [reply ci/dcos.sc]: I'm not ignore this... I think we may need this in other projects and I would like to reduce the number of files we need to link to... leaving this here encapsulates all of the registry tool work to this file.
